python/hsfs/core/statistics_engine.py

```diff
@@ -0,0 +1,39 @@
+#
+#   Copyright 2020 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+import datetime
+
+from hsfs import engine, statistics
+from hsfs.core import statistics_api
+
+
+class StatisticsEngine:
+    def __init__(self, feature_store_id, entity_type):
+        self._statistics_api = statistics_api.StatisticsApi(
+            feature_store_id, entity_type
+        )
+
+    def compute_statistics(self, metadata_instance, feature_dataframe):
+        """Compute statistics for a dataframe and send the result json to Hopsworks."""
+        commit_str = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
```

Comment:
I think for the moment it's ok, but as when we port Hudi, the commit id should come as parameter to the method call.

Reply:
totally agree, commit id should be generated by hudi and passed here. I made sure it's the same format for now.